plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy

```diff
@@ -335,8 +337,7 @@ class AzBatchService implements Closeable {
         final containerOpts = new TaskContainerSettings()
                 .withImageName(container)
                 // mount host certificates otherwise `azcopy fails
-                .withContainerRunOptions('-v /etc/ssl/certs:/etc/ssl/certs:ro -v /etc/pki:/etc/pki:ro')
-
+                .withContainerRunOptions('-v /etc/ssl/certs:/etc/ssl/certs:ro -v /etc/pki:/etc/pki:ro -v /mnt/resource/batch/tasks/fsmounts:/mnt/resource/batch/tasks/fsmounts:rw -e AZ_BATCH_NODE_MOUNTS_DIR=/mnt/resource/batch/tasks/fsmounts')
```

Comment:
This seems hardcoded. What if no Azure share is mounted? Also why the mount path is not configurable? I would expect to be able to mount it in the container into an arbitrary path?

Reply:
This is an excellent catch because I forgot about this point. At the time I submitted the PR, I thought that was the only mount dir. After some experience and a better read of the documentation, I realized that there are different mount points. Specifically:

* `/mnt/resource/batch/tasks/fsmounts` on CentOS nodes
* `/mnt/batch/tasks/fsmounts` on Ubuntu nodes

So, the mount point should definitely go in the config file. For that, what do you think should be the best.

1. This configuration:
```
        storage {
          accountName = ''
          accountKey = ''
          fileName = '<my File Share Name>'
          relativeMountPath = '<my relative Path>'
          rootMountPath = '/mnt/resource/batch/tasks/fsmounts' // or another path, depending on the selected instance type
        }
```

2. Or this:

```
        storage {
          accountName = ''
          accountKey = ''
          fileName = '<my File Share Name>'
          mountPath = '/mnt/resource/batch/tasks/fsmounts/<my relative Path>' // or another path, depending on the selected instance type
```
Even if the second case, it's easier and more readable, I'm leaning for the first one because it lets to define the AZ_BATCH_NODE_MOUNTS_DIR variable in the docker container without manipulating the mount path. Defining this variable is important for portability, because it is available on the node to scripts that do not run inside a container.
I would like to hear your opinion before coding anything here (of course, any other suggestion is more than welcome!). In both cases, we can let Azure Batch returns the error is misconfigured.

Do note that we cannot mount the File Shares under an arbitrary path simply because it's Azure Batch that mounts it. We can only control the relative path.

Regarding your question what happens if no file share is mounted, the answer is... nothing. the dir referred AZ_BATCH_NODE_MOUNTS_DIR exists even if no File Share is mounted. So, worst case, we mount an empty folder.